src/SWP/Bundle/ContentBundle/Service/RouteToArticleMapper.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Content Bundle.
+ *
+ * Copyright 2016 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2016 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\ContentBundle\Service;
+
+use Doctrine\ODM\PHPCR\DocumentRepository;
+use Doctrine\ORM\EntityRepository;
+use Psr\Log\LoggerInterface;
+use SWP\Bundle\ContentBundle\Model\ArticleInterface;
+use SWP\Bundle\ContentBundle\Model\RouteToArticle;
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+
+class RouteToArticleMapper
+{
+    /**
+     * @var EntityRepository
+     */
+    private $routeToArticleRepository;
+
+    /**
+     * @var DocumentRepository
+     */
+    private $routeRepository;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var ExpressionLanguage
+     */
+    private $language;
+
+    public function __construct(EntityRepository $routeToArticleRepository, DocumentRepository $routeRepository, LoggerInterface $logger)
+    {
+        $this->routeToArticleRepository = $routeToArticleRepository;
+        $this->routeRepository = $routeRepository;
+        $this->logger = $logger;
+        $this->language = new ExpressionLanguage();
```

Comment:
You can create a new class `RuleMatcher` and put there logic for matching the rules, register it as a service and inject it here. What if in the future we will want to change the expression language component to something else? You should not change this class code but rather add new class with your own matcher.